openmeter/productcatalog/subscription/http/create.go

```diff
@@ -72,6 +75,13 @@ func (h *handler) CreateSubscription() CreateSubscriptionHandler {
 					return CreateSubscriptionRequest{}, fmt.Errorf("failed to map timing: %w", err)
 				}
 
+				ref := ref.IDOrKey{
+					ID: lo.FromPtrOr(parsedBody.CustomerId, ""),
+				}
+				if ref.ID == "" {
+					ref.Key = lo.FromPtrOr(parsedBody.CustomerKey, "")
+				}
```

Comment:
We could just create the ref here.

```suggestion
				ref := ref.IDOrKey{
					ID: lo.FromPtrOr(parsedBody.CustomerId, ""),
					Key: lo.FromPtrOr(parsedBody.CustomerKey, ""),
				}
```

Reply:
invalid if both are provided

Reply:
They both default to empty strings either way.

Reply:
nvm, i can't read

Reply:
nvm, I can read, it's not equivalent as API supports providing both fields with documented priority for ID while `ref.IDOrKey` only accepts one at a time

Reply:
We talked with @hekike that it might be better if we don't introduce the IDOrKey concept to the package internals so it was changed so that the handler resolves key => id